test/kvao/_helpers.js

```diff
@@ -3,11 +3,22 @@
 var Promise = require('bluebird');
 
 exports.givenCacheItem = function(dataSourceFactory) {
-  var dataSource = dataSourceFactory();
-  return dataSource.createModel('CacheItem', {
+  const dataSource = dataSourceFactory();
+  const CacheItem = dataSource.createModel('CacheItem', {
     key: String,
     value: 'any',
   });
+  if ('deleteAll' in dataSource.connector)
+    CacheItem.deleteAll();
```

Comment:
You need to wait until `deleteAll` returns before starting the tests.

I think we need to change `givenCacheItem` to return a promise now.

```js
exports.givenCacheItem = function(dataSourceFactory) {
  // note - most of this code should be moved to `givenModel`
  // ...
  const p = 'deleteAll' in dataSource.connector ? 
    CacheItem.deleteAll() : Promise.resolve;

    return p.then(() => CacheItem);
};
```